packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -249,6 +249,41 @@ class RenderEditable extends RenderBox {
 
   Rect _lastCaretRect;
 
+  /// Track the visbility of the start of the selected text.
+  ///
+  /// For example, if the text contains "Hello World", and the user selects
+  /// "Hello", then scrolls so only "World" is visible, this will become
+  /// 'false'. If the user scrolls back so that the "H" is visible again, this
+  /// will become 'true'.
+  final ValueNotifier<bool> selectionStartInView = ValueNotifier<bool>(true);
+
+  /// Track the visbility of the end of the selected text.
+  ///
+  /// For example, if the text contains "Hello World", and the user selects
+  /// "World", then scrolls so only "Hello" is visible, this will become
+  /// 'false'. If the user scrolls back so that the "d" is visible again, this
+  /// will become 'true'.
+  final ValueNotifier<bool> selectionEndInView = ValueNotifier<bool>(true);
```

Comment:
we should probably add a paragraph to these two defining "visibility" as meaning "scrolled so that the selection is within the text field viewport", as opposed to being literally visible on the screen.

Reply:
Done.